kv/etcd/store.go

```diff
@@ -220,27 +261,22 @@ func (c *client) tryCleanUp(key string) bool {
 	return true
 }
 
-func (c *client) processNotification(r clientv3.WatchResponse, w kv.ValueWatchable, key string) {
-	err := r.Err()
-	if err != nil {
-		c.logger.Errorf("received error on watch channel: %v", err)
-		c.m.etcdWatchError.Inc(1)
-	}
-
+func (c *client) updateWithRetry(w kv.ValueWatchable, key string) error {
+	var (
+		newValue kv.Value
+		err      error
+	)
 	// we need retry here because if Get() failed on an watch update,
 	// it has to wait 10 mins to be notified to try again
-	if err = c.retrier.Attempt(func() error {
-		return c.update(w, key)
-	}); err != nil {
-		c.logger.Errorf("received notification for key %s, but failed to get value: %v", key, err)
-	}
-}
-
-func (c *client) update(w kv.ValueWatchable, key string) error {
-	newValue, err := c.Get(key)
-	if err == kv.ErrNotFound {
-		// nothing to update
-		return nil
+	if execErr := c.retrier.Attempt(func() error {
+		newValue, err = c.Get(key)
+		if err == kv.ErrNotFound {
+			// do not retry on ErrNotFound
+			return nil
```

Comment:
Good call, will fix, I reopened the pr in https://github.com/m3db/m3cluster/pull/77 btw